src/main/java/org/waarp/openr66/protocol/networkhandler/NetworkServerInitializer.java

```diff
@@ -49,23 +49,21 @@ public NetworkServerInitializer(boolean server) {
     @Override
     protected void initChannel(SocketChannel ch) throws Exception {
         final ChannelPipeline pipeline = ch.pipeline();
+        pipeline.addLast(TIMEOUT, new IdleStateHandler(0, 0,
+                    Configuration.configuration.getTIMEOUTCON(),
+                    TimeUnit.MILLISECONDS));
+        // Global limitation
+        pipeline.addLast(LIMITGLOBAL,
+                Configuration.configuration.getGlobalTrafficShapingHandler());
```

Comment:
This never returns null ?

Reply:
In the current implementation this shouldn't return null.
But maybe a better solution would be to implement `getGlobalTrafficShapingHandler` as a "singleton" getter:
`
public synchronized GlobalTrafficStapingHandler getGlobalTrafficShapingHandler() {
    if (globalTrafficShapingHandler == null) {
        globalTrafficShapingHandler = new GlobalTrafficShapingHandler(x, y, z);
    }
    return globalTrafficShapingHandler;
}
`

Reply:
hum, `synchronized` will sync all threads trying to connect at the same time, which could be not the correct behavior... (less efficiency in multithtread environment)
so make sure the `getGlobalTrafficShapingHandler()` never returns null (initialization should not allow null traffic handler